test/recover_errors_lexer_test.dart

```diff
@@ -110,6 +112,39 @@ void main() {
   suffixProperty();
 }
 
+void beforeInterpolation() {
+  test('should resolve: dangling mustacheEnd at start', () {
+    var results = tokenize('}} some text');
+    expect(results, [
+      new NgToken.interpolationStart(0), // Synthetic
+      new NgToken.interpolationValue(0, ''), // Synthetic
+      new NgToken.interpolationEnd(0),
+      new NgToken.text(2, ' some text'),
+    ]);
+    expect(recoveringException.exceptions.length, 1);
+    var e = recoveringException.exceptions[0];
+    expect(e.context, '}}');
+    expect(e.offset, 0);
+
+    expect(untokenize(results), '{{}} some text');
+  });
+
+  test('should resolve: dangling mustacheEnd at end of text', () {
+    var results = tokenize('mustache text}}');
+    expect(results, [
+      new NgToken.interpolationStart(0), // Synthetic
+      new NgToken.interpolationValue(0, 'mustache text'),
+      new NgToken.interpolationEnd(13),
+    ]);
+    expect(recoveringException.exceptions.length, 1);
+    var e = recoveringException.exceptions[0];
+    expect(e.context, '}}');
+    expect(e.offset, 13);
+
+    expect(untokenize(results), '{{mustache text}}');
```

Comment:
This is a good question. We can try and recover using the preceding text, which is something what current angular_analyzer does, or treat it explicitly as a dangling and insert an open. 

I think the former is a safer 'catch all'. In the case that the preceding text is not a valid expression (due to mistyped expression or meant to be text), it will continue placing the 'mustache text' chunk as both a text as well as a potential mustache. It would give us more flexibility by allowing mustache. 

On the other hand, the latter would only try and autocomplete expression if the cursor is right before the dangling close.